crates/wadm/src/server/handlers.rs

```diff
@@ -1107,6 +1107,16 @@ mod test {
                 .to_string()
                 .contains("The following capability component(s) are missing from the manifest: ")),
         }
+
+        let manifest = deserialize_yaml("../../tests/fixtures/manifests/duplicate_links.yaml")
+            .expect("Should be able to parse");
+
+        match validate_manifest(&manifest).await {
+            Ok(()) => panic!("Should have detected duplicate links"),
+            Err(e) => assert!(e
+                .to_string()
+                .contains("Duplicate link found inside component")),
+        }
```

Comment:
```suggestion
        assert!(e.is_err_and(|e| e.to_string().contains("Duplicate link found inside component"));
```

This could be simplified a bit with https://doc.rust-lang.org/std/result/enum.Result.html#method.is_err_and

Reply:
Since the other sub-tests in this function follow a similar syntax I think I'll leave it as is.